EarlGrey/Core/GREYAutomationSetup.m

```diff
@@ -45,37 +45,29 @@ - (void)perform {
   [selfClass grey_setupCrashHandlers];
   // Force software keyboard.
   [[UIKeyboardImpl sharedInstance] setAutomaticMinimizationEnabled:NO];
-  // Showing accessibility inspector turns on accessibility for the AUT.
-  [selfClass grey_setEnableAccessibilityInspector:YES];
+  // Enable accessibility API.
+  [selfClass grey_setEnableAccessibility];
   // Turn off auto correction as it inteferes with typing on iOS8.2+.
   if (iOS8_2_OR_ABOVE()) {
     [selfClass grey_modifyKeyboardSettings];
   }
-
-  // Setup performed aboves leaves unwanted traces (such as the accessibility inspector)
-  // lingering around after all tests finish.
-  // Register to perform cleanup task right after the first test case ends. After much trial and
-  // error, this approach seems to work nicer than atexit and xctestobservation. atexit isn't
-  // reliable and gets called multiple times and using xctestobservation mysteriously eats up some
-  // of the logs.
-  [[NSNotificationCenter defaultCenter] addObserver:self
-                                           selector:@selector(grey_tearDown:)
-                                               name:kGREYXCTestCaseInstanceDidFinish
-                                             object:nil];
 }
 
 #pragma mark - Automation Setup
 
-// Enables (or disables) the accessibility inspector which is required for using any properties of
-// the accessibility tree.
-+ (void)grey_setEnableAccessibilityInspector:(BOOL)enabled {
-  NSString *accessibilitySettingsPrefBundlePath =
-      @"/System/Library/PreferenceBundles/AccessibilitySettings.bundle/AccessibilitySettings";
-  NSString *accessibilityControllerClassName = @"AccessibilitySettingsController";
-  id accessibilityControllerInstance =
-      [self grey_settingsClassInstanceFromBundleAtPath:accessibilitySettingsPrefBundlePath
-                                         withClassName:accessibilityControllerClassName];
-  [accessibilityControllerInstance setAXInspectorEnabled:@(enabled) specifier:nil];
+// Enables accessibility, which is required for using any properties of the accessibility tree.
++ (void)grey_setEnableAccessibility {
+  CFPreferencesSetAppValue((CFStringRef)@"ApplicationAccessibilityEnabled",
```

Comment:
There doesn't seem to be. The code that handles keyboard settings is fairly complicated, and while it may also use CFPreferences internally to set some values, it does a lot of other work as well. Using a higher-level private API that performs that work, the way EarlGrey does it today, seems best. If there are no issues with the code, I'd recommend not changing it.